### DISCUSSION TITLE
x-model.number doesn't work with x-mask:dynamic="$money"

### DISCUSSION THREAD

**Discussion by sukei** [alpinejs/alpine#3233] [7. Bugs (archived)]
Hello,

The directives `x-model.number="var"` and `x-mask:dynamic="$money"` doesn't works well together. I think this is due to the formatted result that is not a "valid" number and break once `parseFloat(value)` is called. The resulting number is then formatted back by the dynamic mask and some information is lost in the process.

```
type 1 -> parseFloat('1') -> '1' -> displays 1
type 100 -> parseFloat('100') -> '100' -> displays 100
type 1,000.00 -> parseFloat('1,000.00') -> '1' -> displays 1
```

**Comment by parasshah195** [comment]
Encountering this issue, making it tedious having to handle type conversion again across all its use in a data component. Can this be looked into please?

**Reply by ekwoka** [reply]
the mask also masks the underlying value, not just what is in the input.

So it needs to be value on both sides.

What you can do is using getter/setters in the data to handle this in the way you need it better.
